code/game/machinery/planet_beacon.dm

```diff
@@ -0,0 +1,216 @@
+#define BEACON_UNANCHORED 1
+#define BEACON_ANCHORED 2
+#define BEACON_IDLE 3
+#define BEACON_ACTIVE 4
+#define BEACON_DISTRESS 5
+
+#define DISTRESS_COOLDOWN 5 MINUTES // May need to be raised/lower in the future
+#define MAX_NAME_LENGTH 32 // How long of a name the planet can be renamed to
+
+/obj/machinery/power/planet_beacon
+	icon = 'icons/obj/machines/beacon.dmi'
+	icon_state = "beacon"
+	name = "EHF point beacon"
+	desc = "A long-range radio beacon for marking planetoids and points of interest for revisiting."
+	density = TRUE
+	anchored = FALSE
+	clicksound = "terminal_type"
+	use_power = NO_POWER_USE
+	idle_power_usage = IDLE_DRAW_MINIMAL
+	active_power_usage = ACTIVE_DRAW_MINIMAL
+	light_power = 1.5
+	var/state = BEACON_UNANCHORED
+	var/preserving = FALSE // Separate from state to prevent it being unpreserved on distress
```

Comment:
```suggestion
  /// Separate from state to prevent it being unpreserved on distress
	var/preserving = FALSE
```

Please see the autodoc guide at: https://github.com/shiptest-ss13/Shiptest/blob/383cba0099e9a2528dcce331a40c4262e9d5cf1d/.github/AUTODOC_GUIDE.md?plain=1#L85